src/Microsoft.Diagnostics.Monitoring.WebApi/Controllers/DiagController.cs

```diff
@@ -528,6 +612,96 @@ private static TimeSpan ConvertSecondsToTimeSpan(int durationSeconds)
                 TimeSpan.FromSeconds(durationSeconds);
         }
 
+        private ProcessKey? GetProcessKeyFromIdentifier(
```

Comment:
I'm not sure I understand what these methods are doing. Couldn't we just add a new `ProcessKey(int? pid, Guid? uid, string name)` ctor and just pass it into the relevant operation?

Reply:
The logic for these three methods is mostly just for comprehensively handling situations where: 1) users provide no identifiers, 2) users provide a single identifier, 3) users provide multiple connected identifiers, or 4) users provide multiple identifiers that are associated with different processes. `GetProcessKeyFromIdentifier` is mostly just a staging area for determining which fields were provided in the request, `GetProcessKey` determines which Process Key to use based on what was provided in the request, and `ProcessInfoIsEquivalent` is used to validate whether the PIDs/UIDs/Names provided by the user belong to the same process. I haven't been sure how much documentation to add in the code to avoid unnecessary/excessive comments, but I can put more documentation in if you think that would help.

Reply:
I think it makes more sense to add the new ctor and then update https://github.com/dotnet/dotnet-monitor/blob/d4999f15c5cf80bfb98872599b0587a6e641938a/src/Microsoft.Diagnostics.Monitoring.WebApi/DiagProcessFilter.cs#L24 to add a filter for each non-null field on ProcessKey (instead of assuming there is only one valid field). This avoids repeated calls to the GetProcessAsync methods, which can be costly (time-wise) in the connect mode for dotnet-monitor.